Remove Arm64-specific jobs in matrix pipelines

dougbu opened the pull request:
- stop using `$(TargetArchitecture)` in Helix.Common.props and Helix.targets
  - add `$(IsArm64HelixQueue)`
- make Helix queue choices more readable
  - use `<Choose/>
  - remove excess `Condition` clauses

nit: remove unused `$(IsRequiredCheck)` and `$(CreateHelixPayload)` properties

dougbu committed:
Remove Arm64-specific jobs in matrix pipelines
- stop using `$(TargetArchitecture)` in Helix.Common.props and Helix.targets
  - add `$(IsArm64HelixQueue)`
- make Helix queue choices more readable
  - use `<Choose/>
  - remove excess `Condition` clauses

nit: remove unused `$(IsRequiredCheck)` and `$(CreateHelixPayload)` properties

HaoK commented on eng/targets/Helix.Common.props:
```diff
@@ -8,57 +8,57 @@
+    at the moment. And, most Windows-specific checks are handled using @(HelixAvailablePlatform) in Helix.props.
+  -->
+  <Choose>
+    <When Condition=" '$(IsHelixDaily)' != 'true' ">
```
We should take this opportunity to rename IsHelixDaily to something more correct and clear to make this more readable

HaoK replied:
Maybe flip it and make this IsPRCheck to be explicit

dougbu replied:
done

HaoK commented on eng/targets/Helix.props:
```diff
@@ -10,14 +9,14 @@
     <IsWindowsHelixQueue>false</IsWindowsHelixQueue>
-    <IsWindowsHelixQueue Condition="$(HelixTargetQueue.Contains('Windows')) or $(HelixTargetQueue.Contains('windows'))">true</IsWindowsHelixQueue>
     <IsMacHelixQueue>false</IsMacHelixQueue>
-    <IsMacHelixQueue Condition="$(HelixTargetQueue.Contains('OSX')) or $(HelixTargetQueue.Contains('macOs'))">true</IsMacHelixQueue>
```
What's going on with this, or are we just removing IsmacHelixQueue?

HaoK commented on eng/targets/Helix.props:
```diff
@@ -26,6 +25,14 @@
 
+  <!-- Override a few properties in innermost (Publish) build for Helix submission. -->
+  <PropertyGroup Condition=" '$(HelixTargetQueue)' != '' ">
+    <HelixTimeout Condition="$(HelixTargetQueue.StartsWith('Windows.10.Amd64'))">00:40:00</HelixTimeout>
```
we should consider getting rid of this difference between queues and just use 45 min universal timeouts

dougbu replied:
Could you explain why❔ I don't recall seeing frequent Helix timeouts and the higher the value here, the longer tests can hang the systems.

This point also doesn't seem like something we should mix into this PR. Suggest you file an issue for later.

HaoK replied:
For the same reason we don't set a custom timeout to the expected running time on every job, consistency is good enough, this difference is just extra complexity.  We could set a timeout of 5 minutes on the majority of our jobs if we really care about low timeouts, the current situation is we probably have a bandaid that was slapped on to just barely not timeout on windows.  This type of thing is just a sign of an issue, but I'm just pointing it out since you moved it so it stood out, we certainly don't have to fix it now

dougbu replied:
After offline discussion, I'm leaving this to you @HaoK. Your consistency argument convinced me and if we can _lower_ the timeout, all the better.

wtgodbe reviewed (approved):
Helix.Common.Props looks much better

dougbu committed:
Rename `$(IsHelixDaily)` ➡️ `$(IsHelixPRCheck)`
- use new property only in ci.yml and quarantined-pr.yml
- flip check in Helix.Common.props

dougbu commented:
I'm not going to rerun `aspnetcore-helix-matrix` or `aspnetcore-quarantined-tests` after this small change. For future reference (well, until the builds are cleaned) successful builds were:
- [`aspnetcore-helix-matrix` build 20211121.6](https://dev.azure.com/dnceng/public/_build/results?buildId=1480397&view=results)
- [`aspnetcore-quarantined-tests` build 20211121.2](https://dev.azure.com/dnceng/public/_build/results?buildId=1480403&view=results)
